### DISCUSSION TITLE
Apache2 Config Suggestion addition to docs

### DISCUSSION THREAD

**Discussion by oliverbob** [open-webui/open-webui#292] [General]
**Is your feature request related to a problem? Please describe.**
For developers who are comfortable with Apache but have difficulty running it with SSL on reverse proxy. Here is the easiest setup:

Clone this rep: [Ollama-Webui](https://github.com/ollama-webui/ollama-webui.git)

`git clone https://github.com/ollama-webui/ollama-webui.git`

`docker build -t ollama-webui .`

```
docker run -d -p 3000:8080 -e OLLAMA_API_BASE_URL=https://example.com/ollama/api --name ollama-webui --restart always ollama-webui
47840bcf80a103da648471ac93b91cd207cf02e5f508c3a3bb0dbf4dfd00fef5
```

**Describe the solution you'd like**

example.com.conf:

```
<VirtualHost *:80>
    ServerName example.com
    ServerAlias www.example.com

    ProxyPass / http://example.com:3000/
    ProxyPassReverse / http://example.com:3000/
    ErrorLog ${APACHE_LOG_DIR}/example_error.log
    CustomLog ${APACHE_LOG_DIR}/examplei_access.log combined
RewriteEngine on
RewriteCond %{SERVER_NAME} =example.com [OR]
RewriteCond %{SERVER_NAME} =www.example.com
RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>
```

sudo a2ensite example.com.conf

Run lets encrypt:
`sudo certbot --apache -d example.com`

example.com-le-ssl.conf
```
<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerName example.com
    ServerAlias www.example.com

    ProxyPass / http://example.com:3000/ nocanon
    ProxyPassReverse / http://example.com:3000/
    ErrorLog ${APACHE_LOG_DIR}/example_error.log
    CustomLog ${APACHE_LOG_DIR}/example_access.log combined
SSLProxyEngine on
RewriteEngine on
SSLCertificateFile /etc/letsencrypt/live/example.com/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/example.com/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```

Reload Apache2:

`sudo systemctl reload apache2`

Make sure that ports 80, 443 and 3000 are whitelisted in ufw as well as publicly visible/unblocked from your ISP or host provider.

**Describe alternatives you've considered**
The Caddy example file can also be used, but make sure it is not conflicting with Apache.

**Additional context**
This assumes Ollama Web UI is installed on a server running Apache2.

Update:

The following is a working configuration in my system

   ```
    ProxyRequests Off
    ProxyPreserveHost On
    ProxyAddHeaders On
    
    ProxyPass / http://example.com:3000/ nocanon
    ProxyPassReverse / http://example.com:3000/
    SSLProxyEngine on

    <Location "/ollama/api">
        ProxyPass http://example.com:11434/api nocanon
        ProxyPassReverse http://example.com:11434/api/
    </Location>

    #set the location for the version: /api/v1
    <Location "/api/v1">
        ProxyPass http://example.com:3000/api/v1 nocanon
        ProxyPassReverse http://example.com:3000/api/v1
    </Location>
```

Enjoy!

**Comment by TheMasterFX** [comment]
In my case it is working too, but the model whitelisting doesn't work anymore for "Users". Since all traffic goes directly to the ollama Endpoint

**Comment by regularRandom** [comment]
My setup for https://myserver.com/ollama/ isn't working and I don't understand why. It works from http://localhost:3000 but not from https://myserver.com/ollama.

Here is a snippet of my httpd.conf:

```
DEFINE ollama_url 127.0.0.1
DEFINE ollama_port 3000

<Location /ollama>
    ProxyPass http://${ollama_url}:${ollama_port} nocanon
    ProxyPassReverse http://${ollama_url}:${ollama_port}

    # Optional: Security headers
    Header always set X-Content-Type-Options "nosniff"
    Header always set X-Frame-Options "SAMEORIGIN"
    Header always set X-XSS-Protection "1; mode=block"
</Location>

```

This is my docker run command for the open-webui:

```
docker run -d -p 3000:8080 \
  -e OLLAMA_BASE_URL=http://localhost:3000/ollama \
  --add-host=host.docker.internal:host-gateway \
  -v open-webui:/app/backend/data \
  --name open-webui \
  --restart always \
  ghcr.io/open-webui/open-webui:main

```

One of the errors states that it can't fetch dynamically imported module:

> https://myserver.com/_app/immutable/entry/start.16ba83a6.js

What I missed?

**Reply by ex-expat** [reply]
did you ever get this resolved? no matter what I do I am unable to load any pages due to the `dynamically imported module` issue.

**Reply by regularRandom** [reply]
No, I didn't. Still use 192.168.0.2:3000 in the local network.
